source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc

```diff
@@ -37,7 +39,12 @@ SPIFFEValidator::SPIFFEValidator(const Envoy::Ssl::CertificateValidationContextC
 
   if (!config->subjectAltNameMatchers().empty()) {
     for (const auto& matcher : config->subjectAltNameMatchers()) {
-      subject_alt_name_matchers_.push_back(Matchers::StringMatcherImpl(matcher));
+      if (matcher.san_type() ==
```

Comment:
If this isn't true, should probably throw to indicate that the config is invalid. I think that's safer than silently discarding some config.

Reply:
That conflicts with how we handle the deprecated field match_subject_alt_names. When encountering the deprecated field, we create a matcher for each of the four types (see CertificateValidationContextConfigImpl::getSubjectAltNameMatchers). I will add a todo to add an exception when the deprecated field is removed.